src/passes/builtinHandler/mathsOperationToFunction.ts

```diff
@@ -41,33 +33,26 @@ export class MathsOperationToFunction extends ASTMapper {
 
   visitBinaryOperation(node: BinaryOperation, ast: AST): void {
     this.commonVisit(node, ast);
+    const isUnchecked = this.inUncheckedBlock;
+    if (
+      !((isUnchecked && (node.operator === '-' || node.operator === '+')) || node.operator === '**')
```

Comment:
It's easier to understand non-negated conditions (especially when it's so complex). Could you modify the condition?